### DISCUSSION TITLE
Face Swap Plans?

### DISCUSSION THREAD

**Discussion by Sirmadsen** [mcmonkeyprojects/SwarmUI#419] [General]
Hi,

Are there any plans to add a face swap option? Something like the feature Fooocus uses would be awesome.

Not sure if this has been discussed yet, I searched and didn't find any topics on face swapping.

**Comment by mcmonkey4eva** [comment]
There's <https://github.com/mcmonkeyprojects/SwarmUI/blob/master/docs/Features/Prompt%20Syntax.md#automatic-segmentation-and-refining> this
And there's a dedicated FaceTools extension available on the Extensions tab

**Reply by Sirmadsen** [reply]
Thanks, found the extensions tab literally 1 min after posting. *sigh me*

**Comment by Sirmadsen** [comment]
And of course, 1 min after posting this I find the extensions feature with Reactor extension. I do however still think the Fooocus version is way better as it uses a reference, not just a pure swap. With Fooocus face swap you can have items in front of your face without completely ruining the swap.

**Comment by mcmonkey4eva** [comment]
You might be interested in IPAdapter features then - drag an image into the prompt box and configure it at the top left

**Reply by Sirmadsen** [reply]
I'll take a look. Thanks.
